gpx.cc

```diff
@@ -1289,7 +1289,10 @@ void
 GpxFormat::gpx_write_common_extensions(const Waypoint* waypointp, const gpx_point_type point_type) const
 {
   // gpx version we are writing is >= 1.1.
+  garmin_fs_t* gmsd = garmin_fs_t::find(waypointp);
```

Comment:
I am wondering about making this search conditional for efficiency in the normal case, i.e. when not writing garmin extensions, e.g.
  garmin_fs_t* gmsd = opt_garminext? garmin_fs_t::find(waypointp) : nullptr;

Reply:
Here is a way to only search for gmsd when using garmin extensions (as is done in waypt.cc).  Also checks point_type.

```
diff --git a/gpx.cc b/gpx.cc
index 915063c3..90d75613 100644
--- a/gpx.cc
+++ b/gpx.cc
@@ -1289,10 +1289,10 @@ void
 GpxFormat::gpx_write_common_extensions(const Waypoint* waypointp, const gpx_point_type point_type) const
 {
   // gpx version we are writing is >= 1.1.
-  garmin_fs_t* gmsd = garmin_fs_t::find(waypointp);
+  garmin_fs_t* gmsd = nullptr;
 
   if ((opt_humminbirdext && (WAYPT_HAS(waypointp, depth) || WAYPT_HAS(waypointp, temperature))) ||
-      (opt_garminext && gmsd != NULL && gmsd->ilinks != NULL)  ||
+      (opt_garminext && gpxpt_route==point_type && (gmsd = garmin_fs_t::find(waypointp)) && (gmsd->ilinks != nullptr))  ||
       (opt_garminext && gpxpt_waypoint==point_type && (WAYPT_HAS(waypointp, proximity) || WAYPT_HAS(waypointp, temperature) || WAYPT_HAS(waypointp, depth))) ||
       (opt_garminext && gpxpt_track==point_type && (WAYPT_HAS(waypointp, temperature) || WAYPT_HAS(waypointp, depth) || waypointp->heartrate != 0 || waypointp->cadence != 0))) {
     writer->writeStartElement(QStringLiteral("extensions"));
```